### ISSUE TITLE
Add Barrel

### ISSUE THREAD

**Issue by boredhero** [boredhero/pv#6]
* The Barrel is a machine based on a chest for fermenting and aging.

1. The barrel should have a believable texture and *not* the opening-hinge animation of a chest when clicked. 
2. Needs a chest-like GUI
3. Needs to be able to count the passing of time for any Unfermented Agave Wort and Silver Tequila Fifth items in it.
4. After having either of the above items in it for 2 Minecraft days, the Barrel must replace them with Fermented Agave Wort and Gold Tequila Fifth, respectively.

**Comment by Rellit-219** [comment]
How would the interface look? Like chest slots or a fluid tank thing with input/output slots?

**Comment by boredhero** [comment]
> How would the interface look? Like chest slots or a fluid tank thing with input/output slots?

Definitely a chest interface I think. We're putting in an item and putting out an item after a certain period of time. It shouldn't be an overly large chest though.

**Comment by Rellit-219** [comment]
Wouldn't it make more sense to only have one alcoholic drink per barrel with the barrel having fluid in it vs just having bottles in a chest like thing?

**Comment by boredhero** [comment]
> Wouldn't it make more sense to only have one alcoholic drink per barrel with the barrel having fluid in it vs just having bottles in a chest-like thing?

Yes and no. The mash is made by crafting 1 cooked agave with 1 bucket of water which is gonna produce an item that's called unfermented mash that is literally the texture of a bucket with green water and lumps in it. Then that is fermented into an item that is a bucket with brownish chunky water in it. Then that is put into the still, which outputs an empty bucket and a bottle of tequila. 

If we switched to a fluid tank system then we'd need to worry about things like keeping track of "layers" of fermentation because since it takes 2 Minecraft days, what happens if my tank is half full and fermenting and then I want to fill the other half with more unfermented mash? It also adds what from my perspective is the extreme inconvenience to the user of having to empty and fill buckets to move things around. You could add a piping and power system to counteract this, but then this mod gets a little techier than we probably want it to be. Past that point, 1 bottle of tequila is 25 servings.

If a player wants to avoid instadeath, the most they can consume is 9 servings in 1 Minecraft hour + 1 every hour after that, maintaining their standard alcohol units/blood alcohol content at 9. But at that point, they have nausea and blindness effects. So, at max, any 1 player would optimally consume no more than 5-6 in one hour, then 1 once ever Minecraft hour afterward for maintenance purposes.

So, every bottle of tequila, after being crafted into 25 shots, will likely last the player 17.5 real-life minutes of consumption. I'd argue that with a potion that lasts that long, we should make them as onerous and inefficient to produce as possible. Mass production would be useless. At best one players mass production setup could literally be used to supply a server of 100+ players constantly.

**Comment by Rellit-219** [comment]
That makes sense. Ok.

**Comment by boredhero** [comment]
This has sort of been fixed temporarily with a barrel item for the alpha release. But the full thing needs done sooner or later.
